wrapping/PyBuffer.i.in

```diff
@@ -63,12 +83,44 @@
             "Only arrays of 2, 3 or 4 dimensions are supported."
 
         if ( ndarr.ndim == 3 and isVector ) or (ndarr.ndim == 4):
-            imgview = itkPyBuffer@PyBufferTypes@._GetImageFromArray( ndarr, ndarr.shape[-2::-1], ndarr.shape[-1] )
+            imgview = itkPyBuffer@PyBufferTypes@._GetImageViewFromArray( ndarr, ndarr.shape[-2::-1], ndarr.shape[-1] )
         elif ndarr.ndim in ( 2, 3 ):
-            imgview = itkPyBuffer@PyBufferTypes@._GetImageFromArray( ndarr, ndarr.shape[::-1], 1)
+            imgview = itkPyBuffer@PyBufferTypes@._GetImageViewFromArray( ndarr, ndarr.shape[::-1], 1)
 
         return imgview
 
+    GetImageViewFromArray = staticmethod(GetImageViewFromArray)
+
+    def GetImageFromArray(ndarr, isVector=False):
+        """Get an ITK Image of a NumPy array.
+
+        This is a deep copy of the NumPy array buffer and is completely safe without potential
+        side effects.
+
+        If isVector is True, then a 3D array will be treated as a 2D vector image,
+        otherwise it will be treated as a 3D image.
+
+        If the array uses Fortran-order indexing, i.e. i,j,k, the Image Size
+        will have the same dimensions as the array shape. If the array uses
+        C-order indexing, i.e. k,j,i, the image Size will have the dimensions
+        reversed from the array shape.
+
+        Therefore, since the *numpy.transpose* operator on a 2D array simply
+        inverts the indexing scheme, the Image representation will be the
+        same for an array and its transpose. If flipping is desired, see
+        *numpy.reshape*.
+        """
+
+        # perform deep copy of the array buffer
+        array_copy = numpy.array(ndarr)
+
+        image = itkPyBuffer@PyBufferTypes@.GetImageViewFromArray(array_copy, isVector)
+
+        # attaches the copy of the array to the image to avoid releasing memory
+        # when leaving current scope.
+        image.ndarr = array_copy
```

Comment:
@thewtex I used your trick here, to avoid having to modify `GetImageViewFromArray` (same implementation for Vectors and Matrices). What do you think?